preload/preload-referrer-policy.html

```diff
@@ -0,0 +1,123 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>The preload's referrerpolicy attribute should be respected</title>
+<meta name="timeout" content="long">
+<script src="resources/dummy.js?link-header-preload2"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/preload/resources/preload_helper.js"></script>
+<body>
+    <p>The preload's referrerpolicy attribute should be respected,
+    and consumed regardless of consumer referrer policy</p>
+<script>
+window.referrers = {};
+const {REMOTE_ORIGIN} = get_host_info();
+const loaders = {
+    header: async (t, {preloadPolicy, resourcePolicy, href, id}) => {
+        const iframe = document.createElement('iframe');
+        const params = new URLSearchParams();
+        params.set('href', href);
+        params.set('resource-policy', resourcePolicy);
+        if (preloadPolicy === '')
+            params.set('preload-policy', '');
+        else
+            params.set('preload-policy', `referrerpolicy=${preloadPolicy}`);
+        iframe.src = `resources/link-header-referrer-policy.html?${params.toString()}`;
+        t.add_cleanup(() => iframe.remove());
+        const done = new Promise(resolve => {
+            window.addEventListener('message', ({data}) => {
+                if (id in data.referrers)
+                    resolve({actualReferrer: data.referrers[id]});
+            })
+        });
+        document.body.appendChild(iframe);
+        const {actualReferrer} = await done;
+        return {actualReferrer, unsafe: iframe.src};
+    },
+    element: async (t, {preloadPolicy, resourcePolicy, href, id}) => {
+        const link = document.createElement('link');
+        link.href = href;
+        link.as = 'script';
+        link.rel = 'preload';
+        link.referrerPolicy = preloadPolicy;
+        const preloaded = new Promise(resolve => link.addEventListener('load', resolve));
+        t.add_cleanup(() => link.remove());
+        document.head.appendChild(link);
+        await preloaded;
+        const script = document.createElement('script');
+        script.src = href;
+        script.referrerPolicy = resourcePolicy;
+        const loaded = new Promise(resolve => script.addEventListener('load', resolve));
+        document.body.appendChild(script);
+        await loaded;
+        return {unsafe: location.href, actualReferrer: window.referrers[id]}
+    },
+}
+function test_referrer_policy(preloadPolicy, resourcePolicy, crossOrigin, type) {
+    promise_test(async t => {
+        const id = token();
+        const href = `${crossOrigin ? REMOTE_ORIGIN : ''}/preload/resources/echo-referrer.py?uid=${id}`;
+        const {actualReferrer, unsafe} = await loaders[type](t, {preloadPolicy, resourcePolicy, id, href})
```

Comment:
You got the href and the ID inversed here. I wonder how this was still working..

Reply:
They're named (in a `{}`), the order doesn't matter. I'll flip it for readability though

Reply:
Oops, I knew I was missing something.. :)